sstables/sstable_directory.cc

```diff
@@ -276,8 +276,7 @@ future<> sstable_directory::filesystem_components_lister::process(sstable_direct
     //   a separate step to fetch all files, followed by another step to distribute and process.
 
     directory_lister lister(_directory, lister::dir_entry_types::of<directory_entry_type::regular>(), &manifest_json_filter);
-    std::exception_ptr ex;
-    try {
+    co_await with_closeable(std::move(lister), [this] (directory_lister& lister) -> future<> {
         while (true) {
             auto de = co_await lister.get();
```

Comment:
Does the nested lambda not need coroutine::lambda?

It's moved by with_closeable() so may generate a dangling reference.

Reply:
AFAIU it doesn't (and CI doesn't prove otherwise), the lambda is not used as continuation, but it called instantly

Reply:
@avikivity , do you still have concerns about coroutine::lambda?

Reply:
@avikivity , do you still have concerns about coroutine::lambda?

Reply:
Added coroutine::lambda in two places -- those that capture more than just "this"